File: src/main/java/com/saybetter/client/api/RecommendClient.java
package com.saybetter.client.api;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Component;

import com.saybetter.client.api.dto.RecommendRequest;
import com.saybetter.client.api.dto.RecommendResult;
import com.saybetter.client.converter.RecommendResultConverter;

import lombok.RequiredArgsConstructor;

@Component
@RequiredArgsConstructor
public class RecommendClient {

	private final RecommendApi recommendApi;

	@Value("${recommend.api.identity-token}")
	private String identityToken;

	public RecommendResult.SymbolRecommend recommend(String requestParameter) {
		RecommendRequest.SymbolRecommend request = RecommendRequest.SymbolRecommend.builder()
				.name(requestParameter)
				.build();

		return RecommendResultConverter.toSymbolRecommend(
				recommendApi.recommend(identityToken, request)
		);
	}
}

File: clients/client-api/src/main/kotlin/io/say/better/client/symbol/client/RecommendClient.kt
package io.say.better.client.symbol.client

import io.say.better.client.symbol.api.RecommendApi
import io.say.better.client.symbol.converter.RecommendResultConverter
import io.say.better.client.symbol.dto.RecommendRequest
import io.say.better.client.symbol.dto.RecommendResult
import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.PropertySource
import org.springframework.stereotype.Component

@Component
@PropertySource("classpath:client-api.yml")
class RecommendClient internal constructor(
    private val recommendApi: RecommendApi,
    @Value("\${recommend.api.identity-token}") private val identityToken: String,
) {
    fun recommend(requestParameter: String): RecommendResult.SymbolRecommend {
        val request = RecommendRequest.SymbolRecommend(requestParameter)
        return RecommendResultConverter.toSymbolRecommend(recommendApi.recommend(identityToken, request))
    }
}
